Lib9c/BlockPolicySource.cs

```diff
@@ -111,25 +112,40 @@ BlockChain<NCAction> blockChain
                     // The authorization proof has to have no actions at all.
                     return !transaction.Actions.Any();
                 }
-                
+
+                // Check ActivateAccount
                 if (transaction.Actions.Count == 1 &&
                     transaction.Actions.First().InnerAction is ActivateAccount aa)
                 {
                     return blockChain.GetState(aa.PendingAddress) is Dictionary rawPending &&
-                        new PendingActivationState(rawPending).Verify(aa);
+                           new PendingActivationState(rawPending).Verify(aa);
                 }
 
-                if (blockChain.GetState(ActivatedAccountsState.Address) is Dictionary asDict)
+                // Check admin
+                if (blockChain.GetState(Addresses.Admin) is Dictionary rawAdmin
+                    && new AdminState(rawAdmin).AdminAddress.Equals(transaction.Signer))
                 {
-                    IImmutableSet<Address> activatedAccounts =
-                        new ActivatedAccountsState(asDict).Accounts;
-                    return !activatedAccounts.Any() ||
-                        activatedAccounts.Contains(transaction.Signer);
+                    return true;
                 }
-                else
+
+                var value2 = blockChain.GetState(transaction.Signer.Derive(ActivationKey.DeriveKey));
+                switch (value2)
```

Comment:
IMO `value2` isn't needed.

```suggestion
                switch (blockChain.GetState(transaction.Signer.Derive(ActivationKey.DeriveKey)))
```